programs/ssl/ssl_client2.c

```diff
@@ -960,6 +1018,7 @@ static int my_verify( void *data, mbedtls_x509_crt *crt,
     return( 0 );
 }
 
+#if !defined(MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL) 
```

Comment:
Why is this guard being added? Ultimately, we want the example application to be able to serve both TLS 1.2 and TLS 1.3, and i nthis case, `ssl_sig_hashes_for_test[]` would be needed even if `MBEDTLS_SSL_PROTO_TLS1_3_EXPERIMENTAL` is enabled.

I understand that there the prototype does currently not allow for simultaneous enabling of TLS 1.2 and TLS 1.3, but I think we should try to not amplify this gap further, since it's one of the main obstacles towards upstreaming.